src/lib/distributions.ml

```diff
@@ -21,3 +21,31 @@ let standard_normal_pdf =
 
 let poisson_cdf ~mean k =
     Functions.gammaq (floor (k +. 1.0)) mean
+
+let ln_beta_pdf ~alpha ~beta =
+  if alpha <= 0.0 then raise (Invalid_argument "alpha") else
+    if beta <= 0.0 then raise (Invalid_argument "beta") else
+      let alpha_minus_1 = alpha -. 1.0 in
+      let beta_minus_1 = beta -. 1.0 in
+      let z = Functions.ln_beta_func alpha beta in fun x ->
+        if x < 0.0 || x > 1.0 then neg_infinity else
+          if x = 0.0 then
+            (if (alpha < 1.0) then raise (Invalid_argument "alpha") else neg_infinity)
+            else if x = 1.0 then (if (beta < 1.0) then raise (Invalid_argument "beta")
+          else neg_infinity) else
+            let logX = log x in
+            let log1_minus_x = log1p (-.x) in
+            alpha_minus_1 *. logX +. beta_minus_1 *. log1_minus_x -. z
+
+let beta_pdf ~alpha ~beta =
+  let ln_pdf = ln_beta_pdf ~alpha ~beta in
+  fun x -> let value = ln_pdf x in
+  if value = neg_infinity then 0.0 else exp value
+
+let beta_cdf ~alpha ~beta =
```

Comment:
I tried this out and I'm not certain if the implementation is correct (or I'm could be doing something wrong).


First, I tried it here, http://keisan.casio.com/exec/system/1180573226

Then, thinking about the algebra, if `alpha=` and `beta=1` looking at the definition of the [incomplete beta](http://en.wikipedia.org/wiki/Beta_function#Incomplete_beta_function) I believe the cdf reduces to a straight line (`f(x) = x`).
But when I tried:
```OCaml
Array.init 100 (fun x -> float x /. 100.0)
|> Array.map (fun x -> x, Oml.Distributions.beta_cdf ~alpha:1.0 ~beta:1.0 x) 

- : (float * float) array =                                                                        
[| (0., 0.)
 ; (0.01, 0.00990000000000000775)
 ; (0.02, 0.0196)
 ; (0.03, 0.0290999999999999974)
 ; (0.04, 0.0384000000000000036)
 ; (0.05, 0.0475)
 ; (0.06, 0.0563999999999999918)
 ; (0.07, 0.0651)
 ; ....
 ; (0.5, 0.25)
 ; (0.51, 0.7501); 
 ...
```